core/blockchain.go

```diff
@@ -1258,9 +1340,10 @@ func (self *BlockChain) update() {
 	}
 }
 
-func blockErr(block *types.Block, err error) {
+func reportBlock(block *types.Block, err error) {
```

Comment:
A method doc would be nice, also detailing that this should be called "inline" and not on a new goroutine, as it will itself spin up the reporting goroutine internally.